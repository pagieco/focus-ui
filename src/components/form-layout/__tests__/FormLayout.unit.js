import { shallowMount } from '@vue/test-utils';
import { FormLayout } from '../index';

describe('FormGroup', () => {
  it('is a valid vue component', () => {
    expect(FormLayout).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowMount(FormLayout).element).toMatchSnapshot();
  });

  it('wraps each child component into a form-group item element', () => {
    const wrapper = shallowMount(FormLayout, {
      slots: {
        default: '<div>element 1</div><div>element 2</div>',
      },
    });

    expect(wrapper.findAll('.form-group__item').length).toBe(2);
  });
});
