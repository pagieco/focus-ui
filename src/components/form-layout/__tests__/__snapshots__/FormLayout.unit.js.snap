// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup matches the snapshot 1`] = `
<div
  class="form-group"
/>
`;
